presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -194,16 +199,67 @@ public void testEverythingImplemented()
         assertAllMethodsOverridden(SystemAccessControl.class, FileBasedSystemAccessControl.class);
     }
 
+    @Test
+    public void testRefreshing()
+            throws Exception
+    {
+        TransactionManager transactionManager = createTestTransactionManager();
+        AccessControlManager accessControlManager = new AccessControlManager(transactionManager);
+        File configFile = newTemporaryFile();
+        configFile.deleteOnExit();
+        copy(new File(getResourcePath("catalog.json")), configFile);
+
+        accessControlManager.setSystemAccessControl(FileBasedSystemAccessControl.NAME, ImmutableMap.of(
+                SECURITY_CONFIG_FILE, configFile.getAbsolutePath(),
+                SECURITY_REFRESH_PERIOD, "1s"));
+
+        transaction(transactionManager, accessControlManager)
+                .execute(transactionId -> {
+                    accessControlManager.checkCanCreateView(transactionId, alice, aliceView);
+                    accessControlManager.checkCanCreateView(transactionId, alice, aliceView);
+                    accessControlManager.checkCanCreateView(transactionId, alice, aliceView);
+                });
+
+        copy(new File(getResourcePath("security-config-file-with-unknown-rules.json")), configFile);
+        sleep(2000);
```

Comment:
replace with `assertEventually` (or sth like that) with proper timeout (you can also set timeout on the test level)

Reply:
`assertEventually` is not related to this